GoMod: go list -deps error: no matching files found

Hey,

We have an issue with the GoMod Package Manager

```
10:35:45.499 [DefaultDispatcher-worker-1] ERROR org.ossreviewtoolkit.analyzer.managers.GoMod - Resolving GoMod dependencies for path 'go.mod' failed with: 
IOException: Running 'go list -deps -f {{with .Module}}{{.Path}} {{.Version}}{{end}} ./...' in '/builds/some-go-app' failed with exit code 1:
XXXXXX.go:10:12: pattern some.file: no matching files found
```

Content of XXXXXX.go
```go
//go:generate cp ../../../some.file .
//go:embed some.file
```
some.file ist generated during build of another component in the project.

We don't want to run "go generate/go build" as a prequisite for ort. So, are there other ways to get this working?

Cheers,
Marco

--- a/analyzer/src/funTest/assets/projects/synthetic/gomod-dangling-embed-expected-output.yml
+++ b/analyzer/src/funTest/assets/projects/synthetic/gomod-dangling-embed-expected-output.yml
@@ -1,0 +1,23 @@
+---
+project:
+  id: "GoMod::gomod_dangling_embed:<REPLACE_REVISION>"
+  definition_file_path: "analyzer/src/funTest/assets/projects/synthetic/gomod-dangling-embed/go.mod"
+  declared_licenses: []
+  declared_licenses_processed: {}
+  vcs:
+    type: "Git"
+    url: "<REPLACE_URL_PROCESSED>"
+    revision: "<REPLACE_REVISION>"
+    path: "<REPLACE_PATH>"
+  vcs_processed:
+    type: "Git"
+    url: "<REPLACE_URL_PROCESSED>"
+    revision: "<REPLACE_REVISION>"
+    path: "<REPLACE_PATH>"
+  homepage_url: ""
+  scopes:
+  - name: "main"
+    dependencies: []
+  - name: "vendor"
+    dependencies: []
+packages: []

--- a/analyzer/src/funTest/assets/projects/synthetic/gomod-dangling-embed/go.mod
+++ b/analyzer/src/funTest/assets/projects/synthetic/gomod-dangling-embed/go.mod
@@ -1,0 +1,3 @@
+module gomod_dangling_embed
+
+go 1.19

--- a/analyzer/src/funTest/assets/projects/synthetic/gomod-dangling-embed/main.go
+++ b/analyzer/src/funTest/assets/projects/synthetic/gomod-dangling-embed/main.go
@@ -1,0 +1,17 @@
+package main
+
+import (
+    _ "embed"
+    "fmt"
+)
+
+//go:generate touch file.txt
+var(
+    //go:embed file.txt
+    file string
+)
+
+func main() {
+    fmt.Println(file)
+    fmt.Println("hello world")
+}

--- a/analyzer/src/funTest/kotlin/managers/GoModFunTest.kt
+++ b/analyzer/src/funTest/kotlin/managers/GoModFunTest.kt
@@ -82,4 +82,13 @@ class GoModFunTest : StringSpec({
 
         result.toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
     }
+
+    "Source files with dangling embed directives do not yield issues" {
+        val definitionFile = testDir.resolve("gomod-dangling-embed/go.mod")
+        val expectedResultFile = testDir.resolve("gomod-dangling-embed-expected-output.yml")
+
+        val result = create("GoMod").resolveSingleProject(definitionFile)
+
+        result.toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
+    }
 })

--- a/analyzer/src/main/kotlin/managers/GoMod.kt
+++ b/analyzer/src/main/kotlin/managers/GoMod.kt
@@ -100,7 +100,7 @@ class GoMod(
 
     override fun transformVersion(output: String) = output.removePrefix("go version go").substringBefore(' ')
 
-    override fun getVersionRequirement(): RangesList = RangesListFactory.create(">=1.19.0")
+    override fun getVersionRequirement(): RangesList = RangesListFactory.create(">=1.21.1")
 
     override fun mapDefinitionFiles(definitionFiles: List<File>): List<File> =
         definitionFiles.filterNot { definitionFile ->

